applications/FluidDynamicsApplication/custom_elements/symbolic_stokes.cpp

```diff
@@ -0,0 +1,4943 @@
+        rOStream << "with constitutive law " << std::endl;
+        this->GetConstitutiveLaw()->PrintInfo(rOStream);
+    }
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::Calculate(
+    const Variable<Vector >& rVariable,
+    Vector& rOutput,
+    const ProcessInfo& rCurrentProcessInfo )
+{
+    noalias( rOutput ) = ZeroVector( StrainSize );
+
+    if (rVariable == FLUID_STRESS) {
+
+        // creating a new data container that goes out of scope after the function is left
+        TElementData dataLocal;
+
+        // transferring the velocity (among other variables)
+        dataLocal.Initialize(*this, rCurrentProcessInfo);
+
+        Vector gauss_weights;
+        Matrix shape_functions;
+        ShapeFunctionDerivativesArrayType shape_derivatives;
+
+        // computing DN_DX values for the strain rate
+        this->CalculateGeometryData(gauss_weights, shape_functions, shape_derivatives);
+        const unsigned int number_of_gauss_points = gauss_weights.size();
+
+        double sumOfGaussWeights = 0.0;
+
+        for (unsigned int g = 0; g < number_of_gauss_points; g++){
+
+            UpdateIntegrationPointData(dataLocal, g, gauss_weights[g], row(shape_functions, g), shape_derivatives[g]);
+
+            const Vector gauss_point_contribution = dataLocal.ShearStress;
+
+            noalias( rOutput ) += gauss_point_contribution * gauss_weights[g];
+            sumOfGaussWeights += gauss_weights[g];
+        }
+
+        for (unsigned int i = 0; i < StrainSize; i++){
+            rOutput[i] = ( 1.0 / sumOfGaussWeights ) * rOutput[i];
+        }
+
+    } else {
+
+        Element::Calculate(rVariable, rOutput, rCurrentProcessInfo);
+
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Protected operations
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::AddTimeIntegratedSystem(
+    TElementData &rData,
+    MatrixType &rLHS,
+    VectorType &rRHS)
+{
+    this->ComputeGaussPointLHSContribution(rData, rLHS);
+    this->ComputeGaussPointRHSContribution(rData, rRHS);
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::AddTimeIntegratedLHS(
+    TElementData &rData,
+    MatrixType &rLHS)
+{
+    this->ComputeGaussPointLHSContribution(rData, rLHS);
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::AddTimeIntegratedRHS(
+    TElementData &rData,
+    VectorType &rRHS)
+{
+    this->ComputeGaussPointRHSContribution(rData, rRHS);
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::UpdateIntegrationPointData(
+    TElementData& rData,
+    unsigned int IntegrationPointIndex,
+    double Weight,
+    const typename TElementData::MatrixRowType& rN,
+    const typename TElementData::ShapeDerivativesType& rDN_DX) const
+{
+    rData.UpdateGeometryValues(IntegrationPointIndex, Weight, rN, rDN_DX);
+    this->CalculateMaterialResponse(rData);
+}

```

Comment:
Copy and paste from base class. Please remove.

Reply:
done.